x-pack/plugins/ingest_manager/server/services/setup.ts

```diff
@@ -9,16 +9,66 @@ import { CallESAsCurrentUser } from '../types';
 import { agentConfigService } from './agent_config';
 import { outputService } from './output';
 import { ensureInstalledDefaultPackages } from './epm/packages/install';
+import {
+  packageToConfigDatasourceInputs,
+  Datasource,
+  AgentConfig,
+  Installation,
+  Output,
+} from '../../common';
+import { getPackageInfo } from './epm/packages';
+import { datasourceService } from './datasource';
 
 export async function setup(
   soClient: SavedObjectsClientContract,
   callCluster: CallESAsCurrentUser
 ) {
-  await Promise.all([
-    outputService.ensureDefaultOutput(soClient),
-    agentConfigService.ensureDefaultAgentConfig(soClient),
-
+  const [installedPackages, defaultOutput, config] = await Promise.all([
     // packages installed by default
     ensureInstalledDefaultPackages(soClient, callCluster),
+    outputService.ensureDefaultOutput(soClient),
+    agentConfigService.ensureDefaultAgentConfig(soClient),
   ]);
+
+  // ensure default packages are added to the default conifg
+  const configWithDatasource = (await agentConfigService.get(
+    soClient,
+    config.id,
+    true
+  )) as AgentConfig;
+  for (const installedPackage of installedPackages) {
+    const datasource = (configWithDatasource.datasources as Datasource[]).find(
```

Comment:
I'd really like to avoid these `as` casts. They avoid the type safety we get from the compiler.

For example, removing `as AgentConfig` shows we need to account for a `null` `configWithDatasource`. 

`as Datasource[]` seems more involved. The issue as I understand it is that `configWithDatasource.datasources` could be `string[] | Datasource[]` and `string`s don't include the `package` property. I don't know if we need to update types or use some other filters/guards to ensure we have what we expect, but I want to keep the benefits of the compiler whenever possible.